packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,151 @@
+// @ts-check
+import { makeScalarMap } from '@agoric/store';
+import { Far, makeMarshal } from '@endo/marshal';
+
+const { details: X, quote: q } = assert;
+
+const makePresence = (_slot, iface) => {
+  const severed = `SEVERED: ${iface.replace(/^Alleged: /, '')}`;
+  const thing = /** @type {any} */ (Far(severed, {}));
+  return thing;
+};
+
+/**
+ * Make context for exporting wallet data where brands etc. can be recognized by boardId.
+ *
+ * All purses, brands, etc. must be registered before serialization / unserialization.
+ * Neither the serializer nor the unserializer creates new presences.
```

Comment:
This is no longer quite the case. The serializer creates `unknown:${nonce}` slots because we don't pre-register things like the instances in zoe invitation amounts.